sdk_contrib/spring_boot2/src/main/java/io/opentelemetry/contrib/spring/boot/ServiceResourceFactory.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.spring.boot;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import io.opentelemetry.trace.AttributeValue;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.boot.info.BuildProperties;
+import org.springframework.boot.info.GitProperties;
+
+/** Factory bean for service resource labels. */
+class ServiceResourceFactory implements FactoryBean<Resource> {
+
+  private static final String SEMVER = "semver:";
+  private static final String GIT = "git:";
+  private static final Pattern SEMVER_PATTERN =
+      Pattern.compile(
+          "^([0-9]+)\\.([0-9]+)\\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\\.[0-9A-Za-z-]+)*))?"
+              + "(?:\\+[0-9A-Za-z-]+)?$");
```

Comment:
Should we add this check also in the [InstrumentationLibraryInfo](https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/src/main/java/io/opentelemetry/sdk/common/InstrumentationLibraryInfo.java) class?

Reply:
Sounds like a good idea but needs to be in a different pull request